File: StayBookingFrontend/lib/model/update_hotel_request_dto.dart
class UpdateHotelRequestDto {
  UpdateHotelRequestDto({
    required this.name,
    required this.description,
    required this.address,
    required this.city,
    required this.state,
    required this.country,
    required this.pincode,
    required this.rating,
    required this.updatedBy,
  });

  final String name;
  final String description;
  final String address;
  final String city;
  final String state;
  final String country;
  final String pincode;
  final double rating;
  final String updatedBy;

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'description': description,
      'address': address,
      'city': city,
      'state': state,
      'country': country,
      'pincode': pincode,
      'rating': rating,
      'updatedby': updatedBy,
    };
  }
}

File: StayBooking/StayBooking/src/main/java/com/dto/UpdateHotelRequestDto.java
package com.dto;

import lombok.Data;
import org.springframework.web.multipart.MultipartFile;

import java.math.BigDecimal;
import java.util.List;

@Data
public class UpdateHotelRequestDto {
    private String name;
    private String description;
    private String address;
    private String city;
    private String state;
    private String country;
    private String pincode;
    private BigDecimal rating;
    private Boolean replacePhotos;
    private String updatedby;
    private List<MultipartFile> photos;
}
